app/src/app/api/v0/inventory/[inventory]/download/route.ts

```diff
@@ -0,0 +1,333 @@
+    if (!sheet) {
+      throw new createHttpError.BadRequest(
+        `Sheet missing for sector ${sector.referenceNumber}`,
+      );
+    }
+
+    for (const inventoryValue of values) {
+      if (!inventoryValue.gpcReferenceNumber) {
+        throw new createHttpError.BadRequest(
+          `GPC reference number missing for inventory value ${inventoryValue.id}`,
+        );
+      }
+      const refNoColumn = sheet.getColumn("B");
+      const rowIndex = refNoColumn.values.findIndex(
+        (sheetValue) => sheetValue === inventoryValue.gpcReferenceNumber,
+      );
+      if (rowIndex === -1) {
+        throw new createHttpError.BadRequest(
+          `Row for ref no ${inventoryValue.gpcReferenceNumber} not found`,
+        );
+      }
+      // + 1 because it's one based and we want one below the summary row
+      // TODO start storing multiple rows for each activity/ fuel type etc.
+      const row = sheet.getRow(rowIndex + 1);
+      if (!row) {
+        throw createHttpError.BadRequest(
+          `Couldn't find row ${rowIndex + 1} on sheet ${sheetIndex}`,
+        );
+      }
+
+      if (inventoryValue.unavailableReason) {
+        row.getCell("AM").value =
+          notationKeyMapping[inventoryValue.unavailableReason];
+        row.getCell("AQ").value = inventoryValue.unavailableExplanation;
+      } else {
+        row.getCell("AO").value = inventoryValue.dataSource?.notes;
+      }
+
+      // TODO get these as separate values from emissions factor seeder
+      const activityUnit = inventoryValue.activityUnits?.split("/")[0];
+      const emissionsFactorUnit = inventoryValue.activityUnits
+        ?.split(" ")[0]
+        .split("/")[1];
+      const activityType = inventoryValue.activityUnits?.split(" ")[1];
+
+      row.getCell("G").value = activityType;
+      row.getCell("N").value = inventoryValue.activityValue;
+      row.getCell("O").value = activityUnit;
+      row.getCell("P").value = emissionsFactorUnit;
+      row.getCell("S").value = "CO2, CH4, N2O"; // gases for emissions factor
+
+      // TODO add emissions factor to Emissions factors sheet (ID 19)
+      const groupedGases = keyBy(
+        inventoryValue.gasValues,
+        (value) => value.gas!,
+      );
+      row.getCell("V").value =
+        groupedGases.CO2?.emissionsFactor?.emissionsPerActivity;
+      row.getCell("W").value =
+        groupedGases.CH4?.emissionsFactor?.emissionsPerActivity;
+      row.getCell("X").value =
+        groupedGases.N2O?.emissionsFactor?.emissionsPerActivity;
+      // TODO is this correct? Or what's the tCO2e of an emissions factor?
+      row.getCell("Y").value = converKgToTons(inventoryValue.co2eq);
+
+      if (
+        inventoryValue.gasValues.some((gasValue) => gasValue.gasAmount != null)
+      ) {
+        row.getCell("AC").value = "✓";
+        row.getCell("AD").value = converKgToTons(groupedGases.CO2?.gasAmount);
+        // row.getCell("AI").value = converKgToTons(groupedGases.CO2?.gasAmount);
```

Comment:
can we remove the commented lines?

Reply:
It's pretty hard to find all the right columns and rows and document them so I'd rather not until the migration to CIRIS lite is done.